src/connection.rs

```diff
@@ -292,10 +292,15 @@ impl Connection {
         connect: Box<dyn FnOnce(&AMQPUri) -> HandshakeResult + Send + Sync>,
         mut options: ConnectionProperties,
     ) -> Result<Connection> {
-        let executor = options
-            .executor
-            .take()
-            .unwrap_or_else(|| Arc::new(async_global_executor_trait::AsyncGlobalExecutor));
+        let executor = options.executor.take();
+
+        #[cfg(feature = "default-runtime")]
+        let executor =
+            executor.unwrap_or_else(|| Arc::new(async_global_executor_trait::AsyncGlobalExecutor));
```

Comment:
I'd say replace the unwrap_or_else with just an or_else and make the expect below unconditional.
Having as little conditional code as possible is preferable